src/app/dim-ui/WeaponGroupingIcon.tsx

```diff
@@ -0,0 +1,104 @@
+import { TagValue } from '@destinyitemmanager/dim-api-types';
+import TagIcon from 'app/inventory/TagIcon';
+import { AmmoIcon } from 'app/item-popup/ItemPopupHeader';
+import { vaultWeaponGroupingSettingSelector } from 'app/settings/vault-grouping';
+import { VaultGroupIconValue } from 'app/shell/item-comparators';
+import { DestinyAmmunitionType } from 'bungie-api-ts/destiny2';
+import { useSelector } from 'react-redux';
+import ElementIcon from './ElementIcon';
+import { getWeaponTypeSvgIconFromCategoryHashes } from './svgs/itemCategory';
+
+const VALID_AMMO_TYPES: Record<DestinyAmmunitionType, boolean> = {
+  [DestinyAmmunitionType.None]: true,
+  [DestinyAmmunitionType.Primary]: true,
+  [DestinyAmmunitionType.Special]: true,
+  [DestinyAmmunitionType.Heavy]: true,
+  [DestinyAmmunitionType.Unknown]: true,
+};
+
+const VALID_TAGS: Record<TagValue, boolean> = {
+  favorite: true,
+  keep: true,
+  infuse: true,
+  archive: true,
+  junk: true,
+};
+
+export default function WeaponGroupingIcon({
+  iconValue,
+  className,
+}: {
+  iconValue: VaultGroupIconValue;
+  className?: string;
+}) {
+  const vaultWeaponGroupingSetting = useSelector(vaultWeaponGroupingSettingSelector);
+
```

Comment:
Is this necessary? I don't think you're using this component if the setting is disabled.

Reply:
The alternative is to pass the grouping setting in. I thought this was cleaner, but don't mind either way.

Reply:
I don't think you need it at all if you pass in the `iconValue` as a tagged union.

Reply:
Yeah, will remove when I add that tagged union. 😊